components/new-contribution-flow/StepProfile.js

```diff
@@ -1,43 +1,348 @@
-    stepDetails: PropTypes.shape({
-      amount: PropTypes.number,
-      interval: PropTypes.string,
-    }),
+const messages = defineMessages({
+  incognito: { id: 'profile.incognito', defaultMessage: 'Incognito' },
+});
+
+const NewContributionFlowStepProfile = ({
+  LoggedInUser,
+  stepDetails,
+  profiles,
+  defaultSelectedProfile,
+  onChange,
+  data,
+}) => {
+  const intl = useIntl();
+  const [locale, setLocale] = useState('en');
+  const [localProfile, setLocalProfile] = useState(defaultSelectedProfile);
+
+  const renderInfoMessage = amount => {
+    if (amount < 25000) {
+      return (
+        <FormattedMessage
+          id="ContributionFlow.lowestContributionInfoMessage"
+          defaultMessage="Please provide a valid email, we need to send you a receipt for your donation."
+        />
+      );
+    } else if (amount >= 25000 && amount < 500000) {
+      return (
+        <FormattedMessage
+          id="ContributionFlow.lowContributionInfoMessage"
+          defaultMessage="Every donation must be linked to an email account for legal reasons. Please provide a valid email."
+        />
+      );
+    } else if (amount >= 500000) {
+      return (
+        <FormattedMessage
+          id="ContributionFlow.highContributionInfoMessage"
+          defaultMessage="While we'll email you a donation receipt, we are still required to keep an address for donations over $5000 USD."
+        />
+      );
+    }
   };
 
-  renderForm = () => {
-    const { LoggedInUser } = this.props;
-    const { amount, interval } = this.props.stepDetails;
-    const guestLowContribution = !LoggedInUser && amount < 500000;
-    const guestHighContribution = !LoggedInUser && amount > 500000;
+  // for guest users
+  const renderGuestForm = amount => {
+    return (
+      <Container as="fieldset" border="none" width={1} py={3}>
+        <Flex justifyContent="space-between">
+          <Box width={1 / 2} mb={3} mr={1}>
+            <StyledInputField
+              label={
+                amount < 25000 ? (
+                  <FormattedMessage id="Fields.FullName.Optional" defaultMessage="Full name (optional)" />
+                ) : (
+                  <FormattedMessage id="Fields.FullName.Required" defaultMessage="Full name (required)" />
+                )
+              }
+              htmlFor="name"
+            >
+              {inputProps => (
+                <StyledInput
+                  {...inputProps}
+                  defaultValue={data?.name}
+                  placeholder="i.e. Thomas Anderson"
+                  required={amount < 25000 ? false : true}
+                  onChange={e => onChange({ stepProfile: { ...data, [e.target.name]: e.target.value } })}
+                />
+              )}
+            </StyledInputField>
+          </Box>
+          <Box width={1 / 2} mb={3} ml={1}>
+            <StyledInputField
+              label={<FormattedMessage id="Fields.email.required" defaultMessage="Email (required)" />}
```

Comment:
`StyledInputField` will add the `(required)` for you if you specify it:

```suggestion
              label={<FormattedMessage id="Email" defaultMessage="Email" />}
              required={false}
```

Reply:
When I have `required={false}` it displays the 'optional'
<img width="547" alt="Screenshot 2020-07-31 at 12 33 14" src="https://user-images.githubusercontent.com/37520401/89031250-0ee1fd00-d32a-11ea-9367-0c7005b7894d.png">

But when I have `required={true}` or `required` it only displays the field name without adding '(required)' on the end.
<img width="554" alt="Screenshot 2020-07-31 at 12 32 55" src="https://user-images.githubusercontent.com/37520401/89031229-07225880-d32a-11ea-84f3-39871c716676.png">

Reply:
Personally I am fine to use `StyledInputField` as is and only mark the optional fields with `optional` and not add an additional '(required)' to the required fields, but I was only following the Figma designs
<img width="1442" alt="Screenshot 2020-07-31 at 12 44 57" src="https://user-images.githubusercontent.com/37520401/89032048-ad229280-d32b-11ea-9668-eb79b9da0499.png">

Reply:
Indeed, it seems to be a new pattern from design - the previous rule was to only mark optional fields as such. What we could do is to add an option on `StyledInputField` like:

```es6
StyledInputField.propsTypes = {
  /** Set this to true to display a "(Required)" next to the label when the field is required */
  showLabelRequired: PropTypes.bool,
}
```

Then have a conditional like:

```jsx
required === true && showLabelRequired && !isCheckbox ? (
  <Span color="black.500">
    <FormattedMessage
      id="RequiredFieldLabel"
      defaultMessage="{field} (required)"
      values={{ field: labelContent }}
    />
  </Span>
) : ...
```